components/mpas-ocean/src/shared/mpas_ocn_submesoscale_eddies.F

```diff
@@ -0,0 +1,182 @@
+             ocn_submesoscale_init
+
+   real(kind=RKIND) :: tau, LfMin, Ce, dsMax
+
+
+  !***********************************************************************
+
+  contains
+
+  !***********************************************************************
+  !
+  !  routine ocn_submesoscale_compute_velocity
+  !
+  !> \brief   Computes velocity from submesoscale eddies
+  !> \details
+  !>  This routine implements the Fox-Kemper et al 2011 submesoscale 
+  !>  parameterization (https://doi.org/10.1016/j.ocemod.2010.09.002).
+  !>  The transport velocity from the submesoscales is given by a 
+  !>  stream function which is given as
+  !>
+  !>  Psi = C_e*Delta_S/L_f*H^2 int(bGrad_H)/sqrt(f^2+tau^-2)*mu(z)
+  !>
+  !>  here C_e is a specified efficiency, Delta_s is taken as 
+  !>  min(dcEdge,dsMax), H is the mixed layer depth, L_f is the subgrid
+  !>  frontal width that is scaled to the coarse grid, f is the 
+  !>  coriolis parameter, and tau is a time scale specified to prevent 
+  !>  the singularity near the equator.  the horizontal buoyancy gradient
+  !>  is integrated over the mixed layer depth and mu(z) is a non-dimensional
+  !>  shape function that is zero below the mixed layer depth
+  !>
+  !***********************************************************************
+
+  subroutine ocn_submesoscale_compute_velocity()
+
+     !-----------------------------------------------------------------
+     !
+     ! local variables
+     !
+     !-----------------------------------------------------------------
+
+     integer :: k, nEdges, nCells, iCell, iEdge, cell1, cell2, indMLDedge
+
+     real(kind=RKIND),dimension(:),allocatable :: &
+        streamFunction, &
+        zEdge
+
+     real(kind=RKIND) :: mu, zMLD, bvfML, hML, gradBuoyML, ds, Lf, bvfAv, &
+        hAv, bldEdge, ustarEdge
+
+     nEdges = nEdgesHalo(2)
+
+     allocate(streamFunction(nVertLevels+1))
+     allocate(zEdge(nVertLevels+1))
+
+     !$omp parallel
+     !$omp do schedule(runtime) &
+     !$omp private(cell1,cell2,bvfML,zEdge,streamFunction,gradBuoyML,hML,bvfAv, &
+     !$omp         zMLD,Lf,ds,mu,k)
+     do iEdge=1,nEdges
+        cell1 = cellsOnEdge(1,iEdge)
+        cell2 = cellsOnEdge(2,iEdge)
+
+        zEdge(:) = 0.0_RKIND
+        streamFunction(:) = 0.0_RKIND
+        bvfML = 0.0_RKIND
+        gradBuoyML = 0.0_RKIND
+        bvfAv = sqrt(0.5_RKIND*(max(BruntVaisalaFreqTop(minLevelCell(cell1)+1,cell1),1.0E-20_RKIND) + &
+                       max(BruntVaisalaFreqTop(minLevelCell(cell2)+1,cell2),1.0E-20_RKIND)))
+        hML = 0.5_RKIND*layerThickEdgeMean(minLevelEdgeBot(iEdge),iEdge)
+        bvfML = bvfML + hML*bvfAv
+        if (minLevelEdgeTop(iEdge) .ge. 1) then
+           gradBuoyML = hML*gradBuoyEddy(minLevelEdgeTop(iEdge)+1,iEdge)
+           bvfML = hML*bvfAv
+        else
+           cycle
+        end if
+
+        indMLDedge = min(indMLD(cell1),indMLD(cell2))
+        do k = minLevelEdgeBot(iEdge)+2,indMLDedge
```

Comment:
As discussed, I think this loop is off by 1. 
```suggestion
        do k = minLevelEdgeBot(iEdge)+1,indMLDedge
```

Reply:
OK, this is a bit more complicated than that. 
Assuming all indexing starts at 1 (e.g. for minLevelCell, minLevelEdgeBot and minLevelEdgeTop), 
the first calculation in the *updated* loop is for k=2, and returns bvfAv = BruntVaisalaFreqTop(2) -- which is the same as the previous value of bvfAv. So I think it is safe to put the bvfAv calculation on L119 before the bvfML update.

Reply:
Yep you’re right, it needs to move.  I’ll fix it.

Reply:
Done.